### ISSUE TITLE
[V3] Implement new Concept Exercise: error-handling

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1374]
This issue describes how to implement the `error-handling` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how `handle errors` in Go.

## Learning objectives

For example:

handling errors:
- `if err != nil` check
- early `return`/`continue` (extract function if early not possible)
- happy path aligned to the left: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88
- return error or handle it -- not both! (logging is handling!)
- on many functions that return an object and error: check the error, not if the object is nil. Don't use object if error occurred.
- don't attempt to write global error handling
- don't ignore errors (there are very few exceptions to this rule)
- don't panic (extra exercise will handle panics in-depth)

returning errors:
- always last argument (already mentioned in `errors` exercise, maybe repeat?)
- other values should be zero values if `err != nil`

## Out of scope

- error creation (can be used, but is not subject of this exercise)
- panic/recover

## Concepts

- `error-handling`

## Prerequisites

- `errors`
- `nil`

## Resources to refer to

- https://blog.golang.org/error-handling-and-go
- https://gobyexample.com/errors
- https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
